src/plugins/PluginsWrapper.ts

```diff
@@ -17,7 +17,7 @@ export class PluginsWrapper implements Omit<MonitoredPlugin, 'initialize'> {
         this.plugins.forEach(p => p.onFailure(opts));
     }
 
-    async increment(name: string, value?: number, tags?: Record<string, string>): Promise<void> {
+    async increment<T extends string>(name: T, value?: number, tags?: Record<string, string>): Promise<void> {
```

Comment:
```suggestion
    async increment<T extends string = string>(name: T, value?: number, tags?: Record<string, string>): Promise<void> {
```

Otherwise, it's potentially breaking, I think. It may infer the string literal in use.

Can you check if it would break something? If it doesn't resolve this comment